### PR TITLE
Replication cleanup

### PR DESCRIPTION
**Pull request by graetzer**
Spinoff from #5066 to make review easier

- Cleanup replication tests
- Set server role earilier during server bootstrap
- Make UserManager optional (and disable on DBServers and agents)
- Use std::chrono::duration in HeartbeatThread

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/281/

File: arangod/Cluster/HeartbeatThread.cpp [hunk 1]
```diff
@@ -427,23 +423,19 @@ void HeartbeatThread::runSingleServer() {
...
-      if (remain >= 0.5) {
-        usleep(500000);
-        remain -= 0.5;
-      } else {
-        usleep(static_cast<TRI_usleep_t>(remain * 1000.0 * 1000.0));
-        remain = 0.0;
-      }
-    }
-    start = TRI_microtime();
+    auto remain = _interval - (std::chrono::steady_clock::now() - start);
```
Comment (jsteemann) [inline arangod/Cluster/HeartbeatThread.cpp]: Is it possible that `remain` gets negative here?

File: arangod/Cluster/HeartbeatThread.cpp [hunk 2]
```diff
@@ -360,7 +358,7 @@ void HeartbeatThread::runDBServer() {
         break;
       }
 
-      double remain = interval - (TRI_microtime() - start);
+      auto remain = _interval - (std::chrono::steady_clock::now() - start);
```
Comment (jsteemann) [inline arangod/Cluster/HeartbeatThread.cpp]: can `remain` get negative here? can it actually store negative values?

File: js/client/modules/@arangodb/process-utils.js [hunk 3]
```diff
@@ -1043,11 +1043,17 @@ function startInstanceCluster (instanceInfo, protocol, options,
...
+    internal.wait(5.0);
+    let opts = makeAuthorizationHeaders(authOpts);
+    opts['headers']['content-type'] =  'application/json';
+    opts['method'] = 'POST';
+    let reply = download(agencyUrl + '/_api/agency/read',
+    '[["/arango/Plan/AsyncReplication/Leader"]]', opts);
+
     if (!reply.error && reply.code === 200) {
       let res = JSON.parse(reply.body);
+<<<<<<< HEAD
```
Comment (jsteemann) [inline js/client/modules/@arangodb/process-utils.js]: please fix merge conflict!

File: arangod/Cluster/HeartbeatThread.cpp (hunk 2 above)
Comment (graetzer) [reply arangod/Cluster/HeartbeatThread.cpp]: is checked later here

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/324/

**Pull request merged**